### DISCUSSION TITLE
Importing a function downloaded from the community results in a 422 error

### DISCUSSION THREAD

**Discussion by czqczqcg** [open-webui/open-webui#4417] [General]
# Bug Report

## Description

**Bug Summary:**
Importing a function downloaded from the community results in a 422 error.

**Steps to Reproduce:**
go to the workspace/functions page, then import a function downloaded from the community

**Expected Behavior:**
Upload successful

**Actual Behavior:**
"POST /api/v1/functions/create HTTP/1.1" 422 Unprocessable Entity

## Environment

- **Open WebUI Version:** v0.3.11
- **Ollama (if applicable):** 0.3.3

- **Operating System:** windows 10
- **Browser (if applicable):** chrome

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.

## Logs and Screenshots

**Docker Container Logs:**
"POST /api/v1/functions/create HTTP/1.1" 422 Unprocessable Entity

## Installation Method
Docker
![3](https://github.com/user-attachments/assets/446cce33-c734-42aa-90b9-85f2cfd4b462)

**Comment by justinh-rahb** [comment]
It would help to know **which** function failed to import.

**Reply by czqczqcg** [reply]
I tried importing three functions (Visualize Data, Run Code, Google Translate), but they all failed.

**Reply by tjdalsry** [reply]
It's not all in json format

**Reply by YetheSamartaka** [reply]
It also does not work for
https://openwebui.com/t/constliakos/web_search

**Comment by ashakoen** [comment]
I experienced the same, but you can copy the function code and manually create it in your OpenWebUI instance with success.

**Comment by tastech12** [comment]
I have the same problem. Unable to import any function or any tools. Same error in console.
"422 Unprocessable Entity"
Followed by:
"Uncaught (in promise) TypeError: t is not a constructor"

**Comment by jmbenedetto** [comment]
Hello. I have the same error. I compared the json file I get when exporting my functions to json with the one i'm getting from the community. They are not the same. I put them here if anyone wants to take a look. Also, If I try to use the automatic import, I need to fill in the function name and description, and they are (off course) mandatory.
[function-System Message Augmented.json](https://github.com/user-attachments/files/17604627/function-System.Message.Augmented.json)
[function-Perplexity.json](https://github.com/user-attachments/files/17604629/function-Perplexity.json)

**Comment by Ryaang** [comment]
I have the same error. 
![image](https://github.com/user-attachments/assets/0f9dcdc0-96fb-4aa1-b91c-9347c8a23d64)

**Comment by dportabella** [comment]
same issue, when trying to import https://openwebui.com/t/nathanwindisch/bbc_news_feeds

**Comment by sermakarevich** [comment]
same here 
```
INFO:     127.0.0.1:53142 - "POST /api/v1/tools/create HTTP/1.1" 422 Unprocessable Entity
```

tried:
```
function-AutoTool Filter.json
tool-Weather.json
tools-export-1733936589358.json
```

**Comment by pkeffect** [comment]
I am experiencing this using both the cuda and dev-cuda images with locally mounted data.

**Comment by shaybc** [comment]
same here, tried several functions, cannot import, here is the output of one of them (**function-Artifacts V2.json**):

windows cmd shows:
`INFO:     127.0.0.1:51368 - "POST /api/v1/functions/create HTTP/1.1" 422 Unprocessable Entity`

browser console log:
```
Functions.svelte:509:11
XHRPOST
http://localhost:8080/api/v1/functions/create
[HTTP/1.1 422 Unprocessable Entity 17ms]

Object { detail: (3) […] }
index.ts:23:11
Uncaught (in promise) TypeError: t is not a constructor
    se dom.js:1228
    ki Toast.svelte:338
    Dt Toast.svelte:335
    ui Toast.svelte:330
    vi Toast.svelte:304
    jt Component.js:148
    Ci Toast.svelte:179
    Kt Toaster.svelte:216
    qt Toaster.svelte:194
    Wt Toaster.svelte:163
    p Toaster.svelte:161
    dt scheduler.js:119
    _t scheduler.js:79
    promise callback*ut scheduler.js:20
    ht Component.js:81
    ctx Component.js:139
    Zi Toaster.svelte:54
    o index.js:56
    f index.js:69
    r helpers.js:24
    p state.js:8
    r state.js:41
    T state.js:65
    onload Functions.svelte:513
    promise callback*Xn/ge/_.onload Functions.svelte:512
    ge Functions.svelte:507
    le lifecycle.js:105
    le lifecycle.js:104
    _ ConfirmDialog.svelte:41
    m ConfirmDialog.svelte:124
    D dom.js:361
    m ConfirmDialog.svelte:129
    p ConfirmDialog.svelte:63
    dt scheduler.js:119
    _t scheduler.js:79
    promise callback*ut scheduler.js:20
    ht Component.js:81
    ctx Component.js:139
    Ie Functions.svelte:389
    D dom.js:361
    m Functions.svelte:448
    _t Component.js:44
    m Functions.svelte:60
    p +page.svelte:17
    dt scheduler.js:119
    _t scheduler.js:79
    promise callback*ut scheduler.js:20
    ht Component.js:81
    ctx Component.js:139
    tl +page.svelte:17
    o index.js:56
    tl +page.svelte:11
    tl +page.svelte:11
    J utils.js:41
    _t Component.js:47
    _t scheduler.js:99
    promise callback*ut scheduler.js:20
dom.js:1228:8

```

when trying to add manually (copy the code from the community into th  '+' add function) i get:

```
Object { detail: "[ERROR: No module named 'open_webui.apps']" }
​
detail: "[ERROR: No module named 'open_webui.apps']"
​
<prototype>: Object { … }
​​
__defineGetter__: function __defineGetter__()
​​
__defineSetter__: function __defineSetter__()
​​
__lookupGetter__: function __lookupGetter__()
​​
__lookupSetter__: function __lookupSetter__()
​​
__proto__: 
​​
constructor: function Object()
​​
hasOwnProperty: function hasOwnProperty()
​​
isPrototypeOf: function isPrototypeOf()
​​
propertyIsEnumerable: function propertyIsEnumerable()
​​
toLocaleString: function toLocaleString()
​​
toString: function toString()
​​
valueOf: function valueOf()
​​
<get __proto__()>: function __proto__()
​​
<set __proto__()>: function __proto__()
index.ts:23:11

```

**Comment by okamototk** [comment]
I reported same error here:
https://github.com/open-webui/open-webui/discussions/8716

Cause is expected json format to import is invalid from downloaded one.

**Comment by AlbertoSinigaglia** [comment]
still no fix? getting the same error trying to import https://openwebui.com/t/justinrahb/image_gen

**Comment by pkeffect** [comment]
Remove all the illegal characters, this will work for any imports that fail for this error.

After clicking GET from the site and its imported into OI it looks like this...

![alt text](https://i.imgur.com/GglcLhD.png)

Remove the illegal characters

![alt text](https://i.imgur.com/N9tNc2W.png)

It should then load the file correctly.

Hope this helps everyone out.
